src/include/function/struct/vector_struct_operations.h

```diff
@@ -10,7 +11,38 @@ struct StructPackVectorOperations : public VectorOperations {
     static std::unique_ptr<FunctionBindData> bindFunc(
         const binder::expression_vector& arguments, FunctionDefinition* definition);
     static void execFunc(const std::vector<std::shared_ptr<common::ValueVector>>& parameters,
-        common::ValueVector& result) {} // Evaluate at compile time
+        common::ValueVector& result) {
+        for (auto i = 0u; i < parameters.size(); i++) {
+            auto& parameter = parameters[i];
+            if (parameter->state == result.state) {
+                continue;
+            }
+            // If the parameter's state is inconsistent with the result's state, we need to copy the
+            // parameter's value to the corresponding child vector.
+            copyParameterValueToStructFieldVector(
+                parameter.get(), common::StructVector::getChildVector(&result, i).get());
+        }
+    }
+    static void copyParameterValueToStructFieldVector(
+        const common::ValueVector* parameter, common::ValueVector* structField) {
+        assert(parameter->state->isFlat());
```

Comment:
comment that if parameter is UnFlat then its state must equal to result state and thus doesn't require copy.

Reply:
done